crates/telio-proxy/src/proxy.rs

```diff
@@ -191,6 +243,40 @@ impl StateEgress {
 
         Ok(())
     }
+
+    fn log_error(err: &Error) {
+        match err {
+            Error::IoError(e) => telio_log_error!("Unable to send. Error {}", e.kind()),
+            _ => telio_log_error!("Unable to send. Error {}", err),
+        }
+    }
+
+    async fn handle_error(&mut self, err: Error) {
+        if self.conn_state != OutboundConnState::Error {
+            Self::log_error(&err);
+
+            match ErrorType::from(err) {
+                ErrorType::RecoverableError => (),
+                ErrorType::UnrecoverableError => {
+                    Self::sleep_forever().await;
```

Comment:
If there's an UnrecoverableError `conn_state` should be `OutboundConnState::Error`, move its assignment to the beginning of this function.